### DISCUSSION TITLE
Howto : ebird as alternative to allaboutbirds

### DISCUSSION THREAD

**Discussion by alexbelgium** [Nachtzuster/BirdNET-Pi#48] [General]
Allaboutbird.org is great and the way it is included as a link to birds page is very nicely done. It works great when DATABASE_LANG is defined on the default `en`

When the DATABASE_LANG is anything else, it also redirects to a google page which is a smart way to still be able to reach the information. However, it is slightly less immersive. Also, some european birds are not in the database which seems more focused on US.

As such, the following code can replace allaboutbirds.org to ebirds.org, which supports more European birds, and allows to translate pages in the language of the user (or reverts to english if not found).
```
#!/bin/bash

# Define variables
HOME="/home/pi"

# Checks structure is correct
if [ ! -d "$HOME"/BirdNET-Pi/model ]; then
    echo "Structure doesn't seem correct, exiting"
    exit 1
fi

# Get the conversion file
curl -f -L -s -S https://raw.githubusercontent.com/alexbelgium/hassio-addons/master/birdnet-pi/rootfs/ebird.txt -o $HOME/BirdNET-Pi/model/ebird.txt
chown pi:pi "$HOME"/BirdNET-Pi/model/ebird.txt

# Adapt the php files
sed -i '/$sciname =/a \\t$ebirdname = shell_exec("grep \\"$( echo \\"$sciname\\" | sed '\''s/_/ /g'\'')\\" /home/pi/BirdNET-Pi/model/ebird.txt | cut -d'\''_'\'' -f2 | sed '\''s/ /_/g'\''");' "$HOME"/BirdNET-Pi/scripts/todays_detections.php
sed -i '/$sciname =/a \\t$ebirdname = shell_exec("grep \\"$( echo \\"$sciname\\" | sed '\''s/_/ /g'\'')\\" /home/pi/BirdNET-Pi/model/ebird.txt | cut -d'\''_'\'' -f2 | sed '\''s/ /_/g'\''");' "$HOME"/BirdNET-Pi/scripts/stats.php
sed -i "s|https://allaboutbirds.org/guide/<?php echo \$comname;?>|https://ebird.org/species/<?php echo \$ebirdname;?>?siteLanguage=${DATABASE_LANG}_${DATABASE_LANG}|g" "$HOME"/BirdNET-Pi/scripts/todays_detections.php
sed -i "s|https://allaboutbirds.org/guide/\$comname|https://ebird.org/species/\$ebirdname?siteLanguage=${DATABASE_LANG}_${DATABASE_LANG}|g" "$HOME"/BirdNET-Pi/scripts/stats.php
```

**Comment by Nachtzuster** [comment]
I'm somewhat annoyed by Allaboutbird.org not being as nice with non-english DB settings too, I like the idea. :+1:

**Comment by alexbelgium** [comment]
Do you want me to try to make a PR so that it is an option from the birdnet.conf? Not sure if i'll succeed but for the moment it works nicely in my addon :)

**Comment by Nachtzuster** [comment]
no, not really, I was going to look into it anyway

**Comment by Nachtzuster** [comment]
@alexbelgium I've merged eBird support, give it a try

**Reply by alexbelgium** [reply]
Thanks i'll try it later today!

**Reply by alexbelgium** [reply]
Works perfectly, thanks very much!

**Comment by Swiftnesses** [comment]
I can't seem to make this stick, clicking update settings reverts to 'allaboutbirds', is there a config entry / change I can add?

**Reply by Swiftnesses** [reply]
Sorted by adding
```
#----------------------  Site to pull info from Images  ------------------------#
## ALLABOUTBIRDS or EBIRD
## default ALLABOUTBIRDS, EBIRD better for eurasian locations
INFO_SITE="EBIRD"
```
to my config... 👍

**Reply by alexbelgium** [reply]
Hi, funny for me it was added automatically... I wonder why not on your system

**Reply by Swiftnesses** [reply]
Likely something funky on my docker tbh (which is a stripped out version of your HA one!).

**Reply by alexbelgium** [reply]
Ah nice you have made a true docker container from my HA one ? Because if it makes your life easier we could try to merge both : even if a HA container cannot be directly used on a normal system, a docker container can be used on HA :) or of course we can keep them separate which avoids potential breakages in the future ;)
